projects/common-platform-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/messaging/Handler.kt

```diff
@@ -54,19 +54,44 @@ class Handler(
             }
 
             // Insert each defendant as a person record
-            val savedEntities = personService.insertPerson(defendant, courtCode)
-
-            notifier.caseCreated(savedEntities.person)
-            savedEntities.address?.let { notifier.addressCreated(it) }
+            val savedEntities = personService.insertPerson(defendant, notification.message.hearing.courtCentre.code)
+
+            val eventName = if (featureFlags.enabled("common-platform-record-creation-toggle")) {
+                notifier.caseCreated(savedEntities.person)
+                savedEntities.address?.let { notifier.addressCreated(it) }
+                "PersonCreated"
+            } else {
+                "SimulatedPersonCreated"
+            }
 
             telemetryService.trackEvent(
-                "PersonCreated", mapOf(
+                eventName,
+                mapOf(
                     "hearingId" to notification.message.hearing.id,
                     "CRN" to savedEntities.person.crn,
                     "personId" to savedEntities.person.id.toString(),
+                    "firstName" to savedEntities.person.forename,
+                    "surname" to savedEntities.person.surname,
+                    "dob" to savedEntities.person.dateOfBirth.toString(),
+                    "genderCode" to savedEntities.person.gender.description,
+                    "pnc" to savedEntities.person.pncNumber.toString(),
```

Comment:
We should be careful not to log PII to application insights, so probably best to stick with the identifiers in here (e.g. CRN / hearing / defendant id).

I know this makes the "dry run" events a bit useless, but we can at least get an idea of how many we would create.  And maybe we could make a call to the prepare a case system to get more details.